src/components/views/BlackCardSelection.js

```diff
@@ -60,6 +76,32 @@ const BlackCardSelection = () => {
 
     // fetch the blackCards from the server (it is the server's responsibility to give us 8 cards)
     useEffect(() => {
+        async function fetchBlackCard() {
+            // get blackCard
+            try {
+                const response = await api.get(`users/${userId}/games/activeGame`,
+                    {
+                        // reconfiguration might be necessary in case token is not in localStorage here
+                        headers: {
+                            "authorization": token
+                        }
+                    });
```

Comment:
I think it is not a problem anymore, the interceptor should work fine. Can you remove any configuration in this file ( + the useLocation statement, just assign ` const userId = localStorage.getItem("id")` , (no need for the token). And if you're at it, you can the state in the login and registration :
![image](https://user-images.githubusercontent.com/75019468/168250353-a0925816-e2cb-4240-98ce-a1982d2bcde4.png)

Reply:
i would suggest we make a new issue for that.
It is false in a lot of views. probably the best to change that all in one. in a branch which is made for that